mlir/lib/Conversion/FuncToLLVM/FuncToLLVM.cpp

```diff
@@ -544,6 +544,12 @@ struct CallOpInterfaceLowering : public ConvertOpToLLVMPattern<CallOpType> {
         callOp.getLoc(), packedResult ? TypeRange(packedResult) : TypeRange(),
         promoted, callOp->getAttrs());
 
+    newOp->setAttr(newOp.getOperandSegmentSizesAttrName(),
+                   rewriter.getDenseI32ArrayAttr(
+                       {static_cast<int32_t>(promoted.size()), 0}));
```

Comment:
Why is this necessary? The builder call should create this automatically from its arguments IIRC (and if it doesn't, then `build` call, if implemented manually, should be updated)

Reply:
The `build` call is calling into an auto-generated overload that builds an operation all from scratch:

```cpp
static void build(
  ::mlir::OpBuilder &,
  ::mlir::OperationState &odsState,
  ::mlir::TypeRange resultTypes,
  ::mlir::ValueRange operands,
  ::llvm::ArrayRef<::mlir::NamedAttribute> attributes = {}
);
```

This overload does not perform any "automatic inference" of operands or attributes or something and we have to set the attribute manually.

The same for the change in `SPIRVToLLVM.cpp`.